runtime/compiler/trj9/arm/codegen/J9TreeEvaluator.cpp

```diff
@@ -0,0 +1,215 @@
+      generateMemSrc1Instruction(cg, ARMOp_strb, node, new (cg->trHeapMemory()) TR::MemoryReference(temp1Reg, temp3Reg, 1, cg), temp2Reg);
+
+      generateLabelInstruction(cg, ARMOp_label, node, noChkLabel, deps);
+      }
+
+   }
+
+void J9::ARM::TreeEvaluator::genWrtbarForArrayCopy(TR::Node *node, TR::Register *srcObjReg, TR::Register *dstObjReg, TR::CodeGenerator *cg)
+   {
+   TR::Compilation * comp = cg->comp();
+   bool ageCheckIsNeeded = false;
+   bool cardMarkIsNeeded = false;
+   TR_WriteBarrierKind gcMode = comp->getOptions()->getGcMode();
+
+   ageCheckIsNeeded = (gcMode == TR_WrtbarOldCheck || gcMode == TR_WrtbarCardMarkAndOldCheck || gcMode == TR_WrtbarAlways);
+   cardMarkIsNeeded = (gcMode == TR_WrtbarCardMark || gcMode == TR_WrtbarCardMarkIncremental);
+
+   if (!ageCheckIsNeeded && !cardMarkIsNeeded)
+      {
+      return;
+      }
+
+   if (ageCheckIsNeeded)
+      {
+      TR::Register *temp1Reg;
+      TR::Register *temp2Reg;
+      TR::RegisterDependencyConditions *conditions;
+
+      TR::LabelSymbol *doneLabel;
+      TR::SymbolReference *wbRef = comp->getSymRefTab()->findOrCreateWriteBarrierBatchStoreSymbolRef(comp->getMethodSymbol());
+
+      if (gcMode != TR_WrtbarAlways)
+         {
+         temp1Reg = cg->allocateRegister();
+         temp2Reg = cg->allocateRegister();
+         doneLabel = generateLabelSymbol(cg);
+
+	 conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(3, 3, cg->trMemory());
+         addDependency(conditions, temp1Reg, TR::RealRegister::NoReg, TR_GPR, cg);
+         addDependency(conditions, temp2Reg, TR::RealRegister::NoReg, TR_GPR, cg);
+
+         TR::Register *metaReg = cg->getMethodMetaDataRegister();
+
+         // temp1Reg = dstObjReg - heapBaseForBarrierRange0
+         generateTrg1MemInstruction(cg, ARMOp_ldr, node, temp1Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(metaReg, offsetof(J9VMThread, heapBaseForBarrierRange0), cg));
+         generateTrg1Src2Instruction(cg, ARMOp_sub, node, temp1Reg, dstObjReg, temp1Reg);
+
+         // if (temp1Reg >= heapSizeForBarrierRage0), object not in the tenured area
+         generateTrg1MemInstruction(cg, ARMOp_ldr, node, temp2Reg,
+               new (cg->trHeapMemory()) TR::MemoryReference(metaReg, offsetof(J9VMThread, heapSizeForBarrierRange0), cg));
+         generateSrc2Instruction(cg, ARMOp_cmp, node, temp1Reg, temp2Reg);
+         generateConditionalBranchInstruction(cg, node, ARMConditionCodeCS, doneLabel);
+         }
+      else
+         {
+	 conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(1, 1, cg->trMemory());
+         }
+
+      addDependency(conditions, dstObjReg, TR::RealRegister::gr0, TR_GPR, cg);
+
+      TR::Instruction *gcPoint = generateImmSymInstruction(cg, ARMOp_bl, node, (uintptr_t)wbRef->getSymbol()->castToMethodSymbol()->getMethodAddress(),
+            new (cg->trHeapMemory()) TR::RegisterDependencyConditions((uint8_t) 0, 0, cg->trMemory()), wbRef);
```

Comment:
What is the purpose of creating an empty register dependency condition as opposed to passing in NULL here?

Reply:
The empty RegisterDependencyConditions came from the code for p.
I confirmed that passing a NULL instead works as expected.